src/parser_cbor.c

```diff
@@ -0,0 +1,2089 @@
+/**
+ * @file parser_cbor.c
+ * @author Meher Rushi <meherrushi2@gmail.com>
+ * @brief CBOR data parser for libyang
+ *
+ * Copyright (c) 2020 - 2023 CESNET, z.s.p.o.
+ *
+ * This source code is licensed under BSD 3-Clause License (the "License").
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://opensource.org/licenses/BSD-3-Clause
+ */
+
+#define _GNU_SOURCE
+
+#ifdef ENABLE_CBOR_SUPPORT
+
+#include <assert.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "compat.h"
+#include "context.h"
+#include "dict.h"
+#include "in_internal.h"
+#include "lcbor.h"
+#include "log.h"
+#include "ly_common.h"
+#include "parser_data.h"
+#include "parser_internal.h"
+#include "plugins_exts.h"
+#include "set.h"
+#include "tree.h"
+#include "tree_data.h"
+#include "tree_data_internal.h"
+#include "tree_schema.h"
+#include "validation.h"
+
+#include <stdio.h>
+#include <string.h>
+void print_json(cbor_item_t *item);
+
+void print_json_string(const cbor_item_t *item)
+{
+    size_t length = cbor_string_length(item);
+    char *str = (char *)cbor_string_handle(item);
+    printf("\"%.*s\"", (int)length, str);
+}
+
+void print_json_array(const cbor_item_t *item)
+{
+    printf("[");
+    size_t size = cbor_array_size(item);
+    cbor_item_t **handle = cbor_array_handle(item);
+
+    for (size_t i = 0; i < size; ++i)
+    {
+        print_json(handle[i]);
+        if (i < size - 1)
+            printf(", ");
+    }
+
+    printf("]");
+}
+
+void print_json_map(const cbor_item_t *item)
+{
+    printf("{");
+    size_t size = cbor_map_size(item);
+    struct cbor_pair *pairs = cbor_map_handle(item);
+
+    for (size_t i = 0; i < size; ++i)
+    {
+        print_json(pairs[i].key);
+        printf(": ");
+        print_json(pairs[i].value);
+        if (i < size - 1)
+            printf(", ");
+    }
+
+    printf("}");
+}
+
+void print_json_number(const cbor_item_t *item)
+{
+    if (cbor_isa_uint(item))
+    {
+        printf("%lu", cbor_get_uint64(item));
```

Comment:
These printf formats tend to throw warnings on some platforms, better to use `PRIu64` and similar.